tests/unittests/sources/test_wsl.py

```diff
@@ -0,0 +1,321 @@
+# Copyright (C) 2024 Canonical Ltd.
+#
+# Author: Carlos Nihelton <carlos.santanadeoliveira@canonical.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import os
+from copy import deepcopy
+from email.mime.multipart import MIMEMultipart
+from pathlib import PurePath
+from typing import cast
+
+from cloudinit import helpers, util
+from cloudinit.sources import DataSourceWSL as wsl
+from tests.unittests.helpers import CiTestCase, mock
+
+INSTANCE_NAME = "Noble-MLKit"
+GOOD_MOUNTS = {
+    "none": {
+        "fstype": "tmpfs",
+        "mountpoint": "/mnt/wsl",
+        "opts": "rw,relatime",
+    },
+    "/dev/sdd": {
+        "fstype": "ext4",
+        "mountpoint": "/",
+        "opts": "rw,relatime,...",
+    },
+    "sysfs": {
+        "fstype": "sysfs",
+        "mountpoint": "/sys",
+        "opts": "rw,nosuid...",
+    },
+    "C:\\": {
+        "fstype": "9p",
+        "mountpoint": "/mnt/c",
+        "opts": "rw,noatime,dirsync,aname=drvfs;path=C:\\;...",
+    },
+    "D:\\": {
+        "fstype": "9p",
+        "mountpoint": "/mnt/d",
+        "opts": "rw,noatime,dirsync,aname=drvfs;path=D:\\;...",
+    },
+    "hugetblfs": {
+        "fstype": "hugetblfs",
+        "mountpoint": "/dev/hugepages",
+        "opts": "rw,relatime...",
+    },
+}
+SAMPLE_LINUX_DISTRO = ("ubuntu", "24.04", "noble")
+
+
+class TestWSLHelperFunctions(CiTestCase):
```

Comment:
As a project, we are trying to move away from CiTestCase wherever possible to make use of pytest constructs, fixtures and features. As such I've adapted your unittests here to move away from the unittest-based self.assert* and tmp directory setup. I have also added unittest coverage for the read_instance_metadata function.